Build failure with http-client 0.5

```
> /tmp/stackage-build8$ stack unpack gogol-0.1.0
Unpacked gogol-0.1.0 to /tmp/stackage-build8/gogol-0.1.0/
> /tmp/stackage-build8/gogol-0.1.0$ runghc -clear-package-db -global-package-db -package-db=/var/stackage/work/builds/nightly/pkgdb Setup configure --package-db=clear --package-db=global --package-db=/var/stackage/work/builds/nightly/pkgdb --libdir=/var/stackage/work/builds/nightly/lib --bindir=/var/stackage/work/builds/nightly/bin --datadir=/var/stackage/work/builds/nightly/share --libexecdir=/var/stackage/work/builds/nightly/libexec --sysconfdir=/var/stackage/work/builds/nightly/etc --docdir=/var/stackage/work/builds/nightly/doc/gogol-0.1.0 --htmldir=/var/stackage/work/builds/nightly/doc/gogol-0.1.0 --haddockdir=/var/stackage/work/builds/nightly/doc/gogol-0.1.0 --flags=
Configuring gogol-0.1.0...
> /tmp/stackage-build8/gogol-0.1.0$ runghc -clear-package-db -global-package-db -package-db=/var/stackage/work/builds/nightly/pkgdb Setup build
Building gogol-0.1.0...
Preprocessing library gogol-0.1.0...
[ 1 of 13] Compiling Network.Google.Internal.Multipart ( src/Network/Google/Internal/Multipart.hs, dist/build/Network/Google/Internal/Multipart.o )

src/Network/Google/Internal/Multipart.hs:17:1: warning: [-Wunused-imports]
    The qualified import of ‘Data.CaseInsensitive’ is redundant
      except perhaps to import instances from ‘Data.CaseInsensitive’
    To import instances alone, use: import Data.CaseInsensitive()
[ 2 of 13] Compiling Network.Google.Compute.Metadata ( src/Network/Google/Compute/Metadata.hs, dist/build/Network/Google/Compute/Metadata.o )

src/Network/Google/Compute/Metadata.hs:229:11: error:
    Not in scope: ‘Client.checkStatus’
    Module ‘Network.HTTP.Conduit’ does not export ‘checkStatus’.
```

--- a/core/gogol-core.cabal
+++ b/core/gogol-core.cabal
@@ -55,7 +55,7 @@ library
         , exceptions           >= 0.6
         , hashable             >= 1.2
         , http-api-data        >= 0.2
-        , http-client          >= 0.4.4
+        , http-client          >= 0.5 && < 1
         , http-media           >= 0.6
         , http-types           >= 0.8.6
         , lens                 >= 4.4

--- a/gogol/gogol.cabal
+++ b/gogol/gogol.cabal
@@ -63,13 +63,12 @@ library
         , conduit              >= 1.1
         , conduit-extra        >= 1.1
         , cryptonite           >= 0.6
-        , data-default-class   >= 0.0.1
         , directory            >= 1.2
         , exceptions           >= 0.6
         , filepath             >= 1.2
         , gogol-core           == 0.1.0.*
-        , http-client          >= 0.4.4
-        , http-conduit         >= 2.1.4
+        , http-client          >= 0.5 && < 1
+        , http-conduit         >= 2.2 && < 3
         , http-media           >= 0.6
         , http-types           >= 0.8.6
         , lens                 >= 4.4

--- a/gogol/src/Network/Google/Compute/Metadata.hs
+++ b/gogol/src/Network/Google/Compute/Metadata.hs
@@ -40,19 +40,20 @@ module Network.Google.Compute.Metadata
     , getMetadata
     ) where
 
+import           Control.Exception       (throwIO)
 import           Control.Monad.Catch
 import           Control.Monad.IO.Class  (MonadIO (..))
 import           Data.Aeson              (eitherDecode')
 import           Data.ByteString         (ByteString)
 import qualified Data.ByteString.Lazy    as LBS
 import           Data.Char               (toLower)
-import           Data.Default.Class      (def)
 import qualified Data.Text.Encoding      as Text
 import qualified Data.Text.Lazy          as LText
 import qualified Data.Text.Lazy.Encoding as LText
 import           Network.Google.Prelude  (Text, (<>))
-import           Network.HTTP.Conduit    (HttpException (..), Manager)
-import qualified Network.HTTP.Conduit    as Client
+import           Network.HTTP.Client     (HttpException (..),
+                                          HttpExceptionContent (..), Manager)
+import qualified Network.HTTP.Client     as Client
 import           Network.HTTP.Types      (HeaderName)
 import           System.Environment      (lookupEnv)
 
@@ -91,7 +92,7 @@ isGCE m = liftIO $ do
     failure = const (pure False)
 
     rq = metadataRequest
-       { Client.responseTimeout = Just 1000000
+       { Client.responseTimeout = Client.responseTimeoutMicro 1000000
        }
 
 -- | A directory of custom metadata values that have been set for this project.
@@ -223,20 +224,24 @@ getMetadata :: MonadIO m
             -> [Int]      -- ^ Acceptable status code responses.
             -> Manager
             -> m (Client.Response LBS.ByteString)
-getMetadata path ss m =
-    liftIO . flip Client.httpLbs m $ metadataRequest
-        { Client.path        = "/computeMetadata/v1/" <> path
-        , Client.checkStatus = \s hs cj ->
-            let c = fromEnum s
-             in if 200 <= c && c < 300 && notElem c ss
-                 then Nothing
-                 else Just . toException $ StatusCodeException s hs cj
-        }
+getMetadata path statuses m =
+    liftIO . flip Client.httpLbs m $
+        metadataRequest
+            { Client.path          = "/computeMetadata/v1/" <> path
+            , Client.checkResponse = \rq rs ->
+                let c = fromEnum (Client.responseStatus rs)
+                 in if 200 <= c && c < 300 && notElem c statuses
+                     then return ()
+                     else do
+                         bs <- Client.brReadSome (Client.responseBody rs) 4096
+                         throwIO . HttpExceptionRequest rq $
+                             StatusCodeException (() <$ rs) (LBS.toStrict bs)
+            }
 
 -- | A default @http-client@ 'Client.Request' with the host, port, and headers
 -- set appropriately for @metadata.google.internal@ use.
 metadataRequest :: Client.Request
-metadataRequest = def
+metadataRequest = Client.defaultRequest
     { Client.host           = "metadata.google.internal"
     , Client.port           = 80
     , Client.secure         = False

--- a/gogol/src/Network/Google/Internal/Auth.hs
+++ b/gogol/src/Network/Google/Internal/Auth.hs
@@ -32,7 +32,6 @@ import           Data.ByteArray.Encoding
 import           Data.ByteString                (ByteString)
 import           Data.ByteString.Builder        ()
 import qualified Data.ByteString.Lazy           as LBS
-import           Data.Default.Class             (def)
 import           Data.String                    (IsString)
 import qualified Data.Text                      as Text
 import qualified Data.Text.Encoding             as Text
@@ -250,11 +249,10 @@ accountsURL :: Text
 accountsURL = "https://accounts.google.com/o/oauth2/v2/auth"
 
 accountsRequest :: Client.Request
-accountsRequest = def
+accountsRequest = Client.defaultRequest
     { Client.host           = "accounts.google.com"
     , Client.port           = 443
     , Client.secure         = True
-    , Client.checkStatus    = \_ _ _ -> Nothing
     , Client.method         = "POST"
     , Client.path           = "/o/oauth2/v2/auth"
     , Client.requestHeaders =

--- a/gogol/src/Network/Google/Internal/HTTP.hs
+++ b/gogol/src/Network/Google/Internal/HTTP.hs
@@ -22,7 +22,6 @@ import           Control.Monad.Trans.Resource      (MonadResource (..))
 import qualified Data.ByteString.Lazy              as LBS
 import           Data.Conduit                      (($$+-))
 import qualified Data.Conduit.List                 as Conduit
-import           Data.Default.Class                (Default (..))
 import           Data.Monoid                       (Dual (..), Endo (..), (<>))
 import qualified Data.Text.Encoding                as Text
 import qualified Data.Text.Lazy                    as LText
@@ -81,11 +80,10 @@ perform Env{..} x = catches go handlers
                 , _serializeBody    = Just bs
                 }
 
-    request ct b = def
+    request ct b = Client.defaultRequest
         { Client.host            = _svcHost
         , Client.port            = _svcPort
         , Client.secure          = _svcSecure
-        , Client.checkStatus     = \_ _ _ -> Nothing
         , Client.responseTimeout = timeout
         , Client.method          = _cliMethod
         , Client.path            = path
@@ -113,7 +111,10 @@ perform Env{..} x = catches go handlers
                     , _serviceBody    = Just b
                     }
 
-    timeout = microseconds <$> _svcTimeout
+    timeout =
+        maybe Client.responseTimeoutNone
+              (Client.responseTimeoutMicro . microseconds)
+              _svcTimeout
 
     handlers =
         [ Handler $ err

--- a/gogol/src/Network/Google/Internal/Logger.hs
+++ b/gogol/src/Network/Google/Internal/Logger.hs
@@ -43,7 +43,6 @@ import           Data.CaseInsensitive         (CI)
 import qualified Data.CaseInsensitive         as CI
 import           Data.Int                     (Int16, Int8)
 import           Data.List                    (intersperse)
-import qualified Data.Text                    as Text
 import qualified Data.Text.Encoding           as Text
 import qualified Data.Text.Lazy               as LText
 import qualified Data.Text.Lazy.Encoding      as LText
@@ -171,7 +170,7 @@ instance ToLog Request where
         ,  "  host      = " <> build (host x) <> ":" <> build (port x)
         ,  "  secure    = " <> build (secure x)
         ,  "  method    = " <> build (method x)
-        ,  "  timeout   = " <> build (responseTimeout x)
+        ,  "  timeout   = " <> build (show (responseTimeout x))
         ,  "  redirects = " <> build (redirectCount x)
         ,  "  path      = " <> build (path            x)
         ,  "  query     = " <> build (queryString     x)

--- a/gogol/src/Network/Google/Internal/Multipart.hs
+++ b/gogol/src/Network/Google/Internal/Multipart.hs
@@ -14,7 +14,6 @@ import           Control.Monad.IO.Class                (MonadIO (..))
 import           Data.ByteString                       (ByteString)
 import qualified Data.ByteString                       as BS
 import           Data.ByteString.Builder.Extra         (byteStringCopy)
-import qualified Data.CaseInsensitive                  as CI
 import           Data.Monoid                           ((<>))
 import           Network.Google.Types                  (Body (..))
 import           Network.HTTP.Client

--- a/script/travis-matrix
+++ b/script/travis-matrix
@@ -43,7 +43,7 @@ ghc_710=( 7.10.3 )
 ghc_801=( 8.0.1 )
 ghc_all=( 8.0.1 7.10.3 )
 
-matrix "Documentation" ghc_710[@] docs[@]
+matrix "Documentation" ghc_801[@] docs[@]
 matrix "Core" ghc_all[@] core[@]
 matrix "Priority" ghc_801[@] priority[@]
 matrix "Remainder" ghc_801[@] library[@]

--- a/stack-7.10.3.yaml
+++ b/stack-7.10.3.yaml
@@ -1,7 +1,10 @@
 compiler:   ghc-7.10.3
-resolver:   lts-6.2
+resolver:   lts-6.9
 flags:      {}
-extra-deps: []
+extra-deps:
+  - http-conduit-2.2.0
+  - http-client-0.5.0.1
+  - http-client-tls-0.3.0
 packages:
   - core
   - gogol

--- a/stack-8.0.1.yaml
+++ b/stack-8.0.1.yaml
@@ -1,11 +1,12 @@
 compiler:   ghc-8.0.1
-resolver:   nightly-2016-06-20
+resolver:   nightly-2016-09-18
 flags:      {}
 extra-deps: []
 packages:
   - core
   - gogol
   - examples
+
   - gogol-adexchange-buyer
   - gogol-adexchange-seller
   - gogol-admin-datatransfer

